suzieq/poller/controller/controller.py

```diff
@@ -0,0 +1,309 @@
+        self._config = defaultdict(lambda: {})
+        self._config.update(config_data.get('poller', {}))
+
+        # Set controller configuration
+        # run_once: ['gather', 'process'] tells the controller if the poller
+        #           should query the devices and exit
+        # period: the update timeout of the inventory
+        # inventory_timeout: the maximum amount of time to wait for an
+        #                    inventory from a source
+        self._run_once = args.run_once
+        self._period = args.update_period or \
+            self._config.get('update-period', 3600)
+        self._inventory_timeout = self._config.get('inventory_timeout', 10)
+
+        # Validate the arguments
+        self._validate_controller_args(args, config_data)
+
+        # Get the inventory
+        inventory_file = args.inventory or self._config['path']
+
+        if inventory_file:
+            if not Path(inventory_file).is_file():
+                raise SqPollerConfError(
+                    f'Inventory file not found at {inventory_file}'
+                )
+        else:
+            raise SqPollerConfError(
+                'No inventory file provided: use -I argument to provide it'
+            )
+
+        source_args = {'run-once': self._run_once,
+                       'path': inventory_file}
+
+        manager_args = {'run-once': self._run_once,
+                        'exclude-services': args.exclude_services,
+                        'no-colescer': args.no_coalescer,
+                        'outputs': args.outputs,
+                        'output-dir': args.output_dir,
+                        'service-only': args.service_only,
+                        'ssh-config-file': args.ssh_config_file,
+                        'config-file': sq_get_config_file(args.config)}
+
+        # Update configuration with command arguments
+        self._config['source'].update(source_args)
+
+        self._config['manager'].update(manager_args)
+        if not self._config['manager'].get('type'):
+            self._config['manager']['type'] = 'static_manager'
+
+        if not self._config['chunker'].get('type'):
+            self._config['chunker']['type'] = 'static_chunker'
+
+    @property
+    def run_once(self) -> str:
+        """Returns the current working mode, wether it is gather or process.
+        If None pollers will run forever.
+
+        Returns:
+            [str]: current run mode
+        """
+        return self._run_once
+
+    @property
+    def period(self) -> int:
+        """Returns the update period of the inventory
+
+        Returns:
+            [int]: update period in seconds
+        """
+        return self._period
+
+    @property
+    def inventory_timeout(self) -> int:
+        """Returns the maximum amount of time to wait for an inventory source
+        retrieving its device list.
+
+        Returns:
+            int: inventory timeout in secodns
+        """
+        return self._inventory_timeout
```

Comment:
Does this need to be inventory source specific? For now, I think not. But don't be surprised if you get asked that

Reply:
There is a timeout for each source which collects the inventory from the source and a single timeout for the controller which collects new inventories from sources and create the global inventory